File: codeforces/ECR26/d.py
#不知为啥老是答案错误，明明是按照C++提交正确的代码改的
n, k = map(int, input().strip().split())
arr = map(int, input().strip().split())
x, y = [],[]
MAXY = 0
def f(xx):
    global MAXY
    xxx = xx
    ans = 0
    while xxx%2 == 0:
        ans += 1
        xxx /= 2
    x.append(ans)
    xxx = xx
    ans = 0
    while xxx%5 == 0:
        ans += 1
        xxx /= 5
    y.append(ans)
    MAXY += ans
dp = [[-1]*4000 for row in range(220)]
for i in arr:
    f(i)
dp[0][0] = 0
for i in range(0,n):
    for j in range(k,0,-1):
        for l in range(MAXY,-1,-1):
            if l-y[i] < 0:
                continue
            elif dp[j-1][l-y[i]] == -1:
                continue
            else:
                dp[j][l] = max(dp[j][l], dp[j-1][l-y[i]]+x[i])
ans = 0
for i in range(1,MAXY+1,1):
    #print(dp[k][i])
    ans = max(ans, min(dp[k][i],i))
print(ans)

File: codeforces/ECR26/d.cpp
#include <iostream>
#include <stdio.h>
#include <string.h>
#include <algorithm>
#include <cmath>
#define ll long long
using namespace std;
struct Nod {
    ll a;
    int x,y;
    Nod(){
        a = x = y = 0;
    }
}nod[220];
void fun(Nod *p) {
    ll x = p->a;
    while(x%2 == 0){
        p->x++;
        x/=2;
    }
    x = p->a;
    while(x%5 == 0){
        p->y++;
        x/=5;
    }
}
int dp[210][18*210];
bool vis[220];
int main() {
    int n, k,MAXY=0;
    cin >> n >> k;
    for(int i = 1; i <= n; i ++)
        cin >> nod[i].a, fun(&nod[i]),MAXY += nod[i].y;
    memset(dp, -1, sizeof(dp));
    // for(int i = 1; i <= n; i ++)printf("%d %d\n",nod[i].x,nod[i].y);
    // printf("%d\n",MAXY);
    dp[0][0] = 0;
    for(int i = 1; i <= n; i ++) {
        for(int j = k; j >= 1; j --) {
            for(int l = MAXY; l >= 0; l --) {
                if(l-nod[i].y < 0) continue;
                if(dp[j-1][l-nod[i].y] == -1) continue;
                dp[j][l] = max(dp[j][l], dp[j-1][l-nod[i].y]+nod[i].x);
            }
        }
    }
    int ans = 0;
    for(int i = 1; i <= MAXY; i ++) {
        ans = max(ans, min(i,dp[k][i]));
    }
    printf("%d\n",ans);
    return 0;
}
